pallets/foreign-investments/src/mock.rs

```diff
@@ -131,6 +129,15 @@ impl pallet_mock_pools::Config for Runtime {
 	type TrancheId = TrancheId;
 }
 
+impl pallet_swaps::Config for Runtime {
+	type Balance = Balance;
```

Comment:
Since we have `BalanceIn` and `BalanceOut` in the Orderbook, shouldn't we also have it in the swaps pallet?

Reply:
That is a good question that I always ask to myself 😆. I'll give another thought.  My last thought was that for the `Swaps` trait has no meaning for it, because the incoming balance is unknown. But for the `pallet-swaps` implementation it can make sense.

Reply:
Okay, I was trying to refactor the pallet using `BalanceIn` and `BalanceOut`, and I think it's not very useful in this case (now). The problem is that `OrderBook` is used in both directions, and we end up with many `.into()` everywhere, losing the "static" benefits of dividing the balances.

So, I would say that in this case, we are making the code "dirty" for almost no static benefit.

Anyway, I still need to think about the `TokenSwaps` trait, which ideally should use the generic in its methods instead of in the trait. Thus, its usage will be much better.  Maybe we can even start using the GATs Rust feature here 🤔. But a task for the future.

Reply:
Wanted to play with GATs, and it seems to work pretty well for this issue: https://github.com/centrifuge/centrifuge-chain/pull/1727